pkg/snapshot/restorer/restorer.go

```diff
@@ -317,25 +326,192 @@ func startEmbeddedEtcd(ro RestoreOptions) (*embed.Etcd, error) {
 	return e, nil
 }
 
-// applyDeltaSnapshot applies the events from time sorted list of delta snapshot to etcd sequentially
+// applyDeltaSnapshots fetches the events from time sorted list of delta snapshots parallelly and applies them to etcd pseudo-sequentially
 func (r *Restorer) applyDeltaSnapshots(client *clientv3.Client, snapList snapstore.SnapList) error {
+	startTime := time.Now()
+	r.logger.Infof("--- APPLY DELTA SNAPSHOTS")
+
+	for _, snap := range snapList {
+		r.logger.Infof("--- APPLY DELTA SNAPSHOTS - snap revision = %d, %d", snap.StartRevision, snap.LastRevision)
+	}
+
 	firstDeltaSnap := snapList[0]
 	if err := r.applyFirstDeltaSnapshot(client, *firstDeltaSnap); err != nil {
 		return err
 	}
 	if err := verifySnapshotRevision(client, snapList[0]); err != nil {
 		return err
 	}
-	for _, snap := range snapList[1:] {
-		if err := r.applyDeltaSnapshot(client, *snap); err != nil {
-			return err
+
+	r.logger.Infof("--- APPLY DELTA SNAPSHOTS - first delta snapshot applied")
+
+	errCh := make(chan error, 1)
+
+	remainingSnaps := snapList[1:]                // stores the snaps, excluding the first one
+	fetchedEventsInfo := make(chan eventsInfo, 1) // channel that passes fetched event data from snaps
+	restoreFinished := make(chan bool, 1)         // channel used to indicate that restoration is finished
+
+	r.logger.Infof("--- APPLY DELTA SNAPSHOTS - channels created")
+
+	// snapLocations is an array to record pointers to temp file locations to which the events objects are stored as json strings.
+	snapLocations := make([]string, len(remainingSnaps))
+
+	go r.fetchListener(snapLocations, fetchedEventsInfo, errCh)
+	r.logger.Infof("--- APPLY DELTA SNAPSHOTS - fetchListener goroutine started")
+	go r.startFetchers(remainingSnaps, snapLocations, fetchedEventsInfo, errCh)
+	r.logger.Infof("--- APPLY DELTA SNAPSHOTS - startFetchers goroutine started")
+	go r.applier(client, remainingSnaps, snapLocations, restoreFinished, errCh)
+	r.logger.Infof("--- APPLY DELTA SNAPSHOTS - applier goroutine started")
+
+	select {
+	case <-restoreFinished:
+		r.logger.Infof("--- APPLY DELTA SNAPSHOTS - select restoreFinished")
+		endTime := time.Now()
+		restorationTime := endTime.Sub(startTime)
+		r.logger.Warnf("RESTORATION TIME: %v", restorationTime)
+		return nil
+	case err := <-errCh:
+		r.logger.Infof("--- APPLY DELTA SNAPSHOTS - select errCh")
+		return err
+	}
+}
+
+// coordinator coordinates the actions of the fetcher routines and the applier routine. It ensures that events are applied in the right order, regardless of the order in which snaps are fetched
```

Comment:
mismatch in function name. `persistIncrSnapshots` would be a better name.

Reply:
Indeed.